### PR TITLE
Fix 2075

### PR DESCRIPTION
**Pull request by Naththo**
#2075

### CODE REVIEW

**Comment by Destroy666x** [comment]
Looks good, but there are 2 problems:
1. Add New Forum is not hidden, the module is wrong.
2. If someone doesn't have permission to see any quick link, the Quick Access menu is still shown. These lines: https://github.com/Naththo/mybb/blob/issue2075/admin/modules/home/module_meta.php#L86-89
should be wrapped by an `if` which should check whether the `$sub_menu` array is filled.

**Comment by Naththo** [comment]
Updated.

File: admin/modules/home/module_meta.php [hunk 1]
```diff
@@ -66,19 +66,30 @@ function home_action_handler($action)
...
+		{
+			if(!check_admin_permissions(array("module" => $sub['module'], "action" => $sub['action']), false))
+			{
+				unset($sub_menu[$id]);
+			}
+		}
+		
+		if(!empty($sub_menu))
+		{
+			$sub_menu = $plugins->run_hooks("admin_home_menu_quick_access", $sub_menu);
```
Comment (Destroy666x) [inline admin/modules/home/module_meta.php]: The hook shouldn't be inside the conditional, plugins may add their own quick links and `$sub_menu` won't be empty then.

**Comment by Naththo** [comment]
Ope, totally missed that.

**Comment by Destroy666x** [comment]
In the last commit you moved `$page->sidebar .= $sidebar->get_markup();` which should be inside the conditional, not the hook

**Comment by Destroy666x** [comment]
Thanks (wrong module is still used for Add New Forum but I'll commit this directly).

**Pull request merged**